horizon/startup/offline_mode.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional, Tuple
+
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.backends import default_backend
+from cryptography.fernet import Fernet
+from pydantic import ValidationError
+import base64
+import secrets
+
+from horizon.startup.schemas import RemoteConfig, RemoteConfigBackup
+from opal_common.logger import logger
+
+
+class OfflineModeManager:
+    """
+    A backup for the remote config, in case the sidecar can't fetch the remote config.
+    """
+
+    def __init__(self, backup_path: str, api_key: str):
+        self._backup_path: str = backup_path
+        self._api_key = api_key
+
+    def _derive_backup_key(self, salt: Optional[bytes] = None) -> Tuple[bytes, bytes]:
+        if salt is None:
+            salt = secrets.token_bytes(16)
```

Comment:
what is secrets ? how can you guarantee that after a restart this will generate the same salt ?

Reply:
1. Library used to generate cryptographic random. You also use it for generating the API key itself on the backend.
2. It wouldn't generate the same salt. When we backup - a salt is randomly generated and stored along side the encrtypted payload. When trying to load from backup - that salt is used to derive the same key from the API key.